### PR TITLE
[NextPluggableDevice]fix NextPluggableDevice allocator(use pjrt allocator) and Sync()

### PR DESCRIPTION
**Pull request by jzhoulon**
use AsyncValueAllocator if context is using PJRT runtime. And fix Sync() to call done callback to avoid hang issue in direct session.

### CODE REVIEW

**Comment by jzhoulon** [comment]
@jyingl3  can you help to take a look? thanks

**Review by jyingl3 — APPROVED** [review]
Thank you for the fix!

**Comment by jzhoulon** [comment]
@jyingl3 I saw the PR is stuck with some internal checks Failed, if there is something need to be fixed in my PR, please let me know, thanks.

**Comment by jyingl3** [comment]
Thanks Zhoulong! Could you update tensorflow/tensorflow/core/common_runtime/next_pluggable_device/BUILD to include tensorflow/core/tfrt/common:async_value_tensor? Sorry not noticing it earlier. Thank you!

**Comment by jzhoulon** [comment]
> Thanks Zhoulong! Could you update tensorflow/tensorflow/core/common_runtime/next_pluggable_device/BUILD to include tensorflow/core/tfrt/common:async_value_tensor? Sorry not noticing it earlier. Thank you!

@jyingl3  done, please have a look. thanks

**Review by jyingl3 — APPROVED** [review]

**Comment by chuanhaozhuge** [comment]
Hi Zhoulong, we are actively testing some TPU implementation that uses NPD to meet deadline and worry this change may cause breakage. Hope you don't mind if we withhold merging this change temporarily.

**Comment by jzhoulon** [comment]
> Hi Zhoulong, we are actively testing some TPU implementation that uses NPD to meet deadline and worry this change may cause breakage. Hope you don't mind if we withhold merging this change temporarily.

sure, we will wait. Thanks

File: tensorflow/core/common_runtime/next_pluggable_device/next_pluggable_device.cc [hunk 1]
```diff
@@ -52,15 +52,19 @@ NextPluggableDevice::NextPluggableDevice(const SessionOptions& session_options,
                                                      options.device_ordinal)),
       device_ordinal_(options.device_ordinal),
       compilation_device_type_(options.compilation_device_name) {
-  allocator_ = std::make_unique<NextPluggableDeviceAllocator>(device_ordinal_);
   if (absl::GetFlag(FLAGS_next_pluggable_device_use_pjrt)) {
```
Comment (jyingl3) [inline tensorflow/core/common_runtime/next_pluggable_device/next_pluggable_device.cc:55]: Do you mind adding a new flag FLAGS_next_pluggable_device_use_pjrt_allocator to guard this behavior?

**Comment by jyingl3** [comment]
Thank you Zhoulong and sorry for the delay. Could you add a new flag so that the allocator can be switched? Left a comment about that. Thank you!!

File: tensorflow/core/common_runtime/next_pluggable_device/next_pluggable_device.cc (hunk 1 above)
Comment (jzhoulon) [reply tensorflow/core/common_runtime/next_pluggable_device/next_pluggable_device.cc]: @jyingl3 done, please have a review. thanks very much.

**Review by jyingl3 — APPROVED** [review]

File: tensorflow/core/common_runtime/next_pluggable_device/next_pluggable_device.cc (hunk 1 above)
Comment (jyingl3) [reply tensorflow/core/common_runtime/next_pluggable_device/next_pluggable_device.cc]: Approved the change. Thank you so much!

**Pull request merged**